Src/DependencyCollector/Shared/Implementation/HttpProcessing.cs

```diff
@@ -0,0 +1,404 @@
+namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+    using System.Linq;
+    using System.Net;
+#if !NET40
+    using System.Web;
+#endif
+    using Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Common;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.DependencyCollector.Implementation.Operation;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+#if NET40
+    using Microsoft.ApplicationInsights.Net40;
+#endif
+    using Microsoft.ApplicationInsights.Web.Implementation;
+
+    /// <summary>
+    /// Concrete class with all processing logic to generate RDD data from the callbacks
+    /// received from Profiler instrumentation for HTTP or HTTP EventSource/DiagnosticSource events.   
+    /// </summary>
+    internal abstract class HttpProcessing
```

Comment:
This is done to minimize changes to the ProfilerHttpProcessing class as much as possible. I believe given the current circumstances, this is probably the best option with lowest chance of regressing tests and functionality and be able to ship on time. If I start refactoring this into a helper class, a whole bunch of references and unit tests would break because the things they reference into ProfilerHttpProcessing have all of a sudden gone missing. To really address it the way you want, much more time is needed with consideration of what our test strategy should be.